### DISCUSSION TITLE
Can slither automatically switch to the correct compiler version?

### DISCUSSION THREAD

**Discussion by izcoser** [crytic/slither#2042] [Q&A] [answered]
Consider I have solc 0.8.0 and 0.7.0 installed, and an input contract requires 0.7.0.
Solc-select is using 0.8.0.

Is there a flag for Slither to infer the required version and switch to 0.7.0 automatically?

**Comment by 0xalpharush** [comment, marked as answer]
There is not currently a flag to do this but there's a relevant open issue that would encompass this https://github.com/crytic/crytic-compile/issues/210

**Comment by izcoser** [comment]
Thank you.
